internal/manifest/manage.go

```diff
@@ -27,18 +34,110 @@ func Create(path string) error {
 	return filesystem.WriteFile(path, formattedInitManifest)
 }
 
-// Load loads a manifest from a JSON file, unmarshals it, and returns a Manifest object.
-func Load(path string) (*Manifest, error) {
+// Loads, validates, and unmarshals the manifest file at the given path into a flagset
+func LoadFlagSet(manifestPath string) (*flagset.Flagset, error) {
 	fs := filesystem.FileSystem()
-	data, err := afero.ReadFile(fs, path)
+	data, err := afero.ReadFile(fs, manifestPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading contents from file %q", manifestPath)
+	}
+
+	validationErrors, err := Validate(data)
+	if err != nil {
+		return nil, err
+	} else if len(validationErrors) > 0 {
+		// TODO tease running manifest validate command
+		return nil, errors.New(FormatValidationError(validationErrors))
+	}
+
+	var flagset flagset.Flagset
+	if err := json.Unmarshal(data, &flagset); err != nil {
+		return nil, fmt.Errorf("error unmarshaling JSON: %v", validationErrors)
+	}
+
+	return &flagset, nil
+}
+
+func Write(path string, flagset flagset.Flagset) error {
+	m := &initManifest{
+		Schema: flagManifestSchemaURL,
+		Manifest: Manifest{
+			Flags: map[string]any{},
+		},
+	}
+	for _, flag := range flagset.Flags {
```

Comment:
It might be more maintainable if the flagset.Flagset type implements MarshallJSON method